api/pkg/s3/bucketlifecycleput.go

```diff
@@ -0,0 +1,136 @@
+// Copyright 2019 The OpenSDS Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package s3
+
+import (
+	"encoding/xml"
+	"net/http"
+
+	"github.com/emicklei/go-restful"
+	"github.com/micro/go-log"
+
+	"crypto/md5"
+
+	"github.com/opensds/multi-cloud/api/pkg/policy"
+
+	"github.com/opensds/multi-cloud/s3/pkg/model"
+	s3 "github.com/opensds/multi-cloud/s3/proto"
+	"golang.org/x/net/context"
+)
+
+func (s *APIService) BucketLifecyclePut(request *restful.Request, response *restful.Response) {
+	if !policy.Authorize(request, response, "bucket:put") {
+		return
+	}
+	bucketName := request.PathParameter("bucketName")
+	log.Logf("Received request for create bucket: %s", bucketName)
+	ctx := context.Background()
+	bucket, _ := s.s3Client.GetBucket(ctx, &s3.Bucket{Name: bucketName})
+	body := ReadBody(request)
+	log.Logf("Body request is %v\n", body)
```

Comment:
you can merge the two logs into one

Reply:
>> res, err := s.s3Client.CreateBucket(ctx, bucket) 
When create lifecycle, only need to update the lifecycle configuration of a bucket, so it's no good to use CreateBucket.